lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -691,7 +691,7 @@ class AxisArtist(martist.Artist):
     is constant) line, ticks, ticklabels, and axis label.
     """
 
-    ZORDER = 2.5
+    zorder = ZORDER = 2.5  # ZORDER is a backcompat alias.
```

Comment:
Is this supposed to be muatble? if so, ZORDER should likely be a property.

Reply:
Well, Line2D.zorder (for example) is a plain attribute which therefore is mutable, even though it probably shouldn't.  If you want to change that we should do it throughout the entire library (and even then I guess someone *may* have a real use case for that), so that's not directly related to the PR at hand.

Reply:
I more thinking of if either zorder or ZORDER are mutated by a user. You'll get an inconsistent state since these are two variables. In particular, if a user has previously changed ZORDER, that will not have any effect anymore after this PR.

Reply:
Fair point.  However it's rather tricky to properly warn on this, because this should be a classproperty (you'd want to mutate it on the class, not on the instance...), but then the deprecated decorator doesn't work on those, and even if we made it work on them, you'd get warnings in the sphinx build (because sphinx will try to access it, trigger the warning, fail the build, yada yada) unless you add more machinery to avoid that.
I also think this change is rather unlikely to be a real problem...

Reply:
Ok.

Reply:
I'm concerned that this leaves only a semblance of back-compatibility (an out of sync read), but it does make this class behave like all of the others (as you can read it, but if you set it it is ignored).

However, this brings the class in-line with all of the other artists where if you set the class level `Artist.zorder` that changes the default of all of the new instances (and the reset of existing ones).  Over all, I think this is the right thing to do.